### PR TITLE
Fixes for ROOT6 for the QA and AOD trains

### PR DESCRIPTION
**Pull request by chiarazampolli**
Several macros to configure the QA train had to be modified to make them compatible with ROOT6.

### CODE REVIEW

**Comment by kharlov** [comment]
+1

**Comment by kharlov** [comment]
+1

**Comment by fprino** [comment]
+1

**Comment by gconesab** [comment]
+1
This commit will not work, the macro doing a separate configuration is not loaded, but I have a way to do this separate configuration without loading the macro that I implemented recently, so I approve and will fix when merged.

**Comment by chiarazampolli** [comment]
Ciao Gustavo,

Which macro exactly? It might be that I overlooked something for ROOT5... 

Chiara

**Comment by mpuccio** [comment]
+1

**Comment by chiarazampolli** [comment]
Ciao @gconesab ,

one clarification: the changes work when you run the processing with what will be committed in AliDPG (I am still working on the macros there). I guess that the macro you referred to is run either in the QA train, or the QAAOD, or AOD train. I checked these, and they all work both under ROOT5 and ROOT6 (minor a casting issue, which has nothing to do with loading macros). This of course using the QA, QAAOD and AOD train macros that I have modified in AliDPG.

Cheers,

Chiara

**Comment by gconesab** [comment]
+1
Ciao @chiarazampolli 
I see, I guess I am confused on when I have to include or load a macro inside a macro. What I have seen when testing them on root6 is that I have to do both so that they compile. The macro in question is AliPhysics/PWGPP/EMCAL/macros/ConfigureEMCALRecoUtils.C, but maybe this is just that I do not include properly the paths to the macros in my environment.
Cheers ... Gustavo

**Comment by chiarazampolli** [comment]
Ciao Gustavo,

The ConfigureEMCALRecoUtils.C macro is loaded in the QAtrain_duo.C steering macro, which is in AliDPG, so you don't see here the changes for that. I will commit those on Monday.

Cheers,

Chiara

**Comment by gconesab** [comment]
+1

**Comment by fprino** [comment]
+1

File: PWGGA/CaloTrackCorrelations/macros/QA/AddTaskCalorimeterQA.C [hunk 1]
```diff
@@ -152,13 +151,16 @@ AliAnalysisTaskCaloTrackCorrelation *AddTaskCalorimeterQA(const char *suffix="de
...
   gROOT->LoadMacro("$ALICE_PHYSICS/PWGPP/EMCAL/macros/ConfigureEMCALRecoUtils.C");
-  ConfigureEMCALRecoUtils(recou,
-                          simulation,
-                          bExotic,
-                          bNonLin,
-                          bEnCalib,
-                          bBadMap,
-                          bTiCalib);  
+
+  TString recouStr = Form("(reinterpret_cast<AliEMCALRecoUtils*>(%p))", recou);
```
Comment (dberzano) [inline PWGGA/CaloTrackCorrelations/macros/QA/AddTaskCalorimeterQA.C]: Ahhh it's precisely `%p` what I was referring to, not `%x` 👍

**Comment by chiarazampolli** [comment]
Ciao @gconesab ,

I have committed a change in the AddTask macro for the EMCAL that calls the ConfigEMCAL... without splitting the loading from the call to the function. Maybe this can be useful for you... I did like this in case you use the AddTask in some lego train, where the splitting of loading and using macros is (most likely) not yet there.

Cheers,

Chiara

File: PWGGA/CaloTrackCorrelations/macros/QA/AddTaskCalorimeterQA.C (hunk 1 above)
Comment (chiarazampolli) [reply PWGGA/CaloTrackCorrelations/macros/QA/AddTaskCalorimeterQA.C]: Yes, much simpler than using strstream...

**Comment by dberzano** [comment]
+test to me it's a "yes" and given that we badly need it in production as soon as possible, I'd say, if @pzhristov does not object, that we give one day to the mentioned persons to have a look and approve, otherwise we'll force-approve it and we'll fix any possible problem in production.

**Comment by fprino** [comment]
+1

**Comment by dberzano** [comment]
+1

**Pull request merged**